content/_code-samples/airgapped-wallet/py/airgapped-wallet-TEST.py

```diff
@@ -0,0 +1,227 @@
+import os
+import base64
+import qrcode
+import platform
+from PIL import Image
+from pathlib import Path, PureWindowsPath, PurePath
+from cryptography.fernet import Fernet
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+from xrpl import wallet
+from xrpl.core import keypairs
+from xrpl.utils import xrp_to_drops
+from xrpl.models.transactions import Payment
+from xrpl.transaction import safe_sign_transaction
+
+
+def create_wallet(silent: False):
+    """
+    Generates a keypair
+    """
+    if not silent:
+        print("1. Generating seed...")
+        seed = keypairs.generate_seed()
+
+        print("2. Deriving keypair from seed...")
+        pub, priv = keypairs.derive_keypair(seed)
+
+        print("3. Deriving classic addresses from keypair..\n")
+        address = keypairs.derive_classic_address(pub)
+        private = keypairs.derive_classic_address(priv)
+
+    else:
+        seed = keypairs.generate_seed()
+        pub, priv = keypairs.derive_keypair(seed)
+        address = keypairs.derive_classic_address(pub)
+        private = keypairs.derive_classic_address(priv)
+ 
+    return address, private, seed
+
+
+def sign_transaction(xrp_amount, destination, ledger_seq, wallet_seq, password):
+    """
+    Signs transaction and returns signed transaction blob in QR code
+    """
+    print("1. Retrieving encrypted private key and salt...")
+    with open(get_path("/WalletTEST/private.txt"), "r") as f:
+        seed = f.read()
+        seed = bytes.fromhex(seed)
+
+    with open(get_path("/WalletTEST/salt.txt"), "rb") as f:
+        salt = f.read()
+
+    print("2. Initializing key...")
+    kdf = PBKDF2HMAC(
+        algorithm=hashes.SHA256(),
+        length=32,
+        iterations=100000,
+        salt=salt
+    )
+
+    key = base64.urlsafe_b64encode(kdf.derive(bytes(password.encode())))
+    crypt = Fernet(key)
+
+    print("3. Decrypting wallet's private key using password")
+    seed = crypt.decrypt(seed)
+
+    print("4. Initializing wallet using decrypted private key")
+    _wallet = wallet.Wallet(seed=seed.decode(), sequence=0)
+
+    validated_seq = ledger_seq
+    _wallet.sequence = wallet_seq
+
+    print("5. Constructing payment transaction...")
+    my_tx_payment = Payment(
+        account=_wallet.classic_address,
+        amount=xrp_to_drops(xrp=xrp_amount),
+        destination=destination,
+        last_ledger_sequence=validated_seq + 100,
+        # +100 to catch up with the ledger when we transmit the signed tx blob to Machine 2
+        sequence=_wallet.sequence,
+        fee="10"
+    )
+
+    print("6. Signing transaction...")
+    my_tx_payment_signed = safe_sign_transaction(transaction=my_tx_payment, wallet=_wallet)
+
+    img = qrcode.make(my_tx_payment_signed.to_dict())
```

Comment:
nit: the img generation should probably be under the below print statement since it's part of that QR code logic